frontend/src/pages/Contracts/Contract/ContractDetails/index.tsx

```diff
@@ -1,50 +1,22 @@
-import React, { useEffect, useState } from "react";
-import { useQuery } from "react-query";
-import { useParams } from "react-router-dom";
-import { GDXAccordion } from "../../../../components/GDXAccordion";
-import { apiAxios } from "../../../../utils";
-import { ContractDetailsSection } from "./ContractDetailsSection";
-import { InvoiceSection } from "./InvoiceSection";
+import { Card, Typography } from "@mui/material";
+import { ReadEditPage } from "components/ReadEditPage";
+import { readFields, editFields, initialValues } from "./fields";
 
 export const ContractDetails = () => {
-  const { contractId } = useParams();
-  const [userHasEditCapability, setEditCapability] = useState(false);
-
-  const getContract = async () => {
-    const contract = await apiAxios().get(`contracts/${contractId}`);
-    contract.data.data.notes = contract.data.data.notes ?? "";
-    contract.data.data.description = contract.data.data.description ?? "";
-    return contract.data;
-  };
-
-  /**
-   * @todo Define a good type. "Any" type temporarily permitted.
-   */
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const contractQuery: any = useQuery(`contract - ${contractId}`, getContract, {
-    refetchOnWindowFocus: false,
-    retryOnMount: false,
-    refetchOnReconnect: false,
-    retry: false,
-    staleTime: Infinity,
-  });
-
-  useEffect(() => {
-    const user = contractQuery?.data?.user;
-    setEditCapability(user && user.capabilities.includes("contracts_update_one"));
-  }, [contractQuery]);
-
   return (
-    <>
-      <GDXAccordion sectionTitle="Contract Details">
-        <ContractDetailsSection
-          query={contractQuery}
-          userHasEditCapability={userHasEditCapability}
-        />
-      </GDXAccordion>
-      <GDXAccordion sectionTitle="Invoice Processing">
-        <InvoiceSection />
-      </GDXAccordion>
-    </>
+    <Card sx={{ padding: "5px" }}>
+      <Typography variant="h5" component="h2">
+        Contract Details
+      </Typography>
+      <ReadEditPage
+        what={{ single: "contract", plural: "contracts" }}
+        readFields={readFields}
+        editFields={editFields}
+        createFormInitialValues={initialValues}
+        editCapability={"contracts_update_one"}
```

Comment:
i feel like you might want to pass roles in here, because, there is a role to read, add and possible delete, which really can't be covered by one role.

Reply:
Makes sense, I'll change that in the next PR.